src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,9 +371,38 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && !string.IsNullOrEmpty(Text) &&
+				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
+			{
+				if (_textBoxView != null)
+				{
+					//store  the value configured on the wrapper _textBoxView to restore if user changes from Wrap to UnWrap
+					_isSingleLine = _textBoxView.IsSingleLine;
+
+					//just change if IsSingleLine equals true, to avoid invoke the propertychange in the wrong way
+					if (_textBoxView.IsSingleLine)
+					{
+						//this must be setted to false to Wrap the text
+						_textBoxView?.SetSingleLine(false);
```

Comment:
Null check is unnecessary

```suggestion
						_textBoxView.SetSingleLine(false);
```